src/Forms/CheckLabel.tsx

```diff
@@ -0,0 +1,134 @@
+import React, {
+  ReactElement,
+  useContext,
+  FunctionComponent,
+} from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { fromTheme } from '../Theme';
+import { POSITION, RequiredSymbol } from './InputLabel';
+
+export interface StyledCheckLabelProps {
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** Specifies the id for the label */
+  htmlFor: string;
+}
+
+export interface StyledCheckLabelTextProps {
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** Used to style label text in a different style if disabled is true */
+  disabled?: boolean;
+}
+
+export interface LabelProps {
+  /** The id of the field tied to this label */
+  htmlFor: string;
+  /** Specifies the label text */
+  label: string;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** If true, the label will be styled to indicate that it labels a required field */
+  isRequired?: boolean;
+  /** Used to style label text in a different style if disabled is true */
+  disabled?: boolean;
+}
+
+const generateGrid = (
+  labelPosition: POSITION,
+  isLabelVisible: boolean
+): string => {
+  if (!isLabelVisible) {
+    return `"input input input"
+            "e e e"`;
+  }
+  if (labelPosition === POSITION.RIGHT) {
+    return `"input label label"
+            ". e e"`;
+  }
+  return `"label input input"
+          ". e e"`;
+};
```

Comment:
We should use consistent `grid-template-area` ids between this and `InputLabel`, so they should either both be `i`, `l`, and `e`, or both be `input`, `label`, and `error`.

Reply:
Ah my bad, originally I was thinking that `grid-area` would control which label (`InputLabel` vs. `Checklabel`) got used. I updated this through b9fd9aa to use `l` and `i` across components instead.